csharp/src/Ice/IncomingResponseFrame.cs

```diff
@@ -32,6 +38,19 @@ public sealed class IncomingResponseFrame
 
         private readonly Communicator _communicator;
 
+        public static IncomingResponseFrame WithVoidReturnValue(Communicator communicator,
+                                                                Protocol protocol,
+                                                                Encoding encoding) =>
+            _cachedVoidReturnValueFrames.GetOrAdd((communicator, protocol, encoding), key =>
+            {
+                var data = new List<ArraySegment<byte>>();
+                var stream = new OutputStream(key.Protocol.GetEncoding(), data, new OutputStream.Position(0, 0));
+                stream.WriteByte((byte)ReplyStatus.OK);
+                _ = stream.WriteEmptyEncapsulation(key.Encoding);
+                Debug.Assert(data.Count == 1);
+                return new IncomingResponseFrame(key.Communicator, key.Protocol, data[0]);
```

Comment:
Wouldn't be simpler to keep the empty payloads and use them here to construct the frame?
```suggestion
                return new IncomingResponseFrame(
                    key.Communicator,
                    key.Protocol, 
                    key.Encoding == OldEncoding ? Ice1Definitions.EmptyResponsePayload : Ice2Definitions.EmptyResponsePayload);
```

Reply:
This wouldn't be correct, and the old Empty payloads only correspond to 2 combinations (out of 4):
- ice2 with 2.0 header and 2.0 empty payload encoding
and
- ice1 with 1.0/1 header and 1.1 empty payload encoding